tests/unit_tests/data_types/test_duration.py

```diff
@@ -62,6 +62,12 @@ def test_duration_parse_str_microseconds() -> None:
     assert duration.elapsed == 100 * 1_000
 
 
+def test_duration_parse_str_compound() -> None:
+    """Test Duration.parse with string input in compound duration."""
+    duration = Duration.parse("1h45m")
+    assert duration.elapsed == 3600 * 1_000_000_000 + 45 * 60 * 1_000_000_000
```

Comment:
I would change this to a more comprehensive test so that everything is covered:
```
duration = Duration.parse("1y2w3d4h5m6s7ms8us9ns")
    assert duration.elapsed == (1 * 365 * 86400 * 1_000_000_000) \
        + (2 * 604800 * 1_000_000_000) \
        + (3 * 86400 * 1_000_000_000) \
        + (4 * 3600 * 1_000_000_000) \
        + (5 * 60 * 1_000_000_000) \
        + (6 * 1_000_000_000) \
        + (7 * 1_000_000) \
        + (8 * 1_000) \
        + 9
```